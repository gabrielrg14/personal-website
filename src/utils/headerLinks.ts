const headerLinks = [
    {
        href: "/about",
        label: "About me",
    },
    {
        href: "/projects",
        label: "Projects",
    },
];
  
export default headerLinks;