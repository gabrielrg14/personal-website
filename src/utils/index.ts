export * from "./links"
export * from "./mediaQuery"
export * from "./repositoryName"
export * from "./techs"
