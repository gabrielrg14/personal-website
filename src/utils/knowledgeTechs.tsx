import {
    Vuedotjs,
    Redux,
    Nodedotjs,
    Express,
    Wordpress,
} from "@styled-icons/simple-icons/";
  
const knownTechs = [
    {
        icon: <Vuedotjs />,
        label: "Vue.js",
    },
    {
        icon: <Redux />,
        label: "Redux",
    },
    {
        icon: <Nodedotjs />,
        label: "Node.js",
    },
    {
        icon: <Express />,
        label: "Express.js",
    },
    {
        icon: <Wordpress />,
        label: "Wordpress",
    },
];
  
export default knownTechs;