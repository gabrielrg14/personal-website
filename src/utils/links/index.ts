export * from "./headerLinks"
export * from "./socialMedias"
