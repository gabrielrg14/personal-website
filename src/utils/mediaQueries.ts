import { generateMedia } from "styled-media-query";

const mediaQueries = generateMedia({
    mobile: "576px",
    tablet: "768px",
    desktop: "1200px",
    largeDesktop: "1400px",
});

export default mediaQueries;