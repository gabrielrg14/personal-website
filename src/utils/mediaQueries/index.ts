import { generateMedia } from "styled-media-query"

export const mediaQueries = generateMedia({
    mobile: "576px",
    tablet: "768px",
    desktop: "1200px",
    largeDesktop: "1400px"
})
