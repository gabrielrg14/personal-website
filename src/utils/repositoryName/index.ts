export const formatRepositoryName = (repositoryName: string) => {
    // Removes hyphen from repository name and puts a blank in its place
    return (
        repositoryName
            .replace(/-/g, " ")
            // Capitalize the first letter of each word that makes up the repository's name
            .replace(/(^\w{1})|(\s+\w{1})/g, (letter) => letter.toUpperCase())
    )
}
