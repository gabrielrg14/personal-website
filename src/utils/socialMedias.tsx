import { 
    Linkedin,
    Github,
} from "@styled-icons/simple-icons/";

const socialMedias = [
    {
        link: "https://www.linkedin.com/in/gabrielrapuccigonzalez/",
        icon: <Linkedin />,
    },
    {
        link: "https://github.com/gabrielrg14",
        icon: <Github />,
    },
];

export default socialMedias;