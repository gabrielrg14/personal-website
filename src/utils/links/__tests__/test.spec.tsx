import { USERNAME } from "common"
import { Linkedin } from "@styled-icons/simple-icons/Linkedin"
import { MarkGithub } from "@styled-icons/octicons/MarkGithub"

import { headerLinks, socialMedias } from "utils"

describe("links", () => {
    describe("headerLinks", () => {
        it("should match snapshot", () => {
            expect(headerLinks).toMatchSnapshot()
        })

        it("should contain skills, experiences and projects links", () => {
            expect(headerLinks).toContainEqual({
                href: "/skills",
                label: "Skills"
            })
            expect(headerLinks).toContainEqual({
                href: "/experiences",
                label: "Experiences"
            })
            expect(headerLinks).toContainEqual({
                href: "/projects",
                label: "Projects"
            })
        })
    })

    describe("socialMedias", () => {
        it("should match snapshot", () => {
            expect(socialMedias).toMatchSnapshot()
        })

        it("should contain LinkedIn and GitHub links", () => {
            expect(socialMedias).toContainEqual({
                name: "LinkedIn",
                link: "https://www.linkedin.com/in/gabrielrapuccigonzalez/",
                icon: <Linkedin />
            })
            expect(socialMedias).toContainEqual({
                name: "GitHub",
                link: `https://github.com/${USERNAME}`,
                icon: <MarkGithub />
            })
        })
    })
})
