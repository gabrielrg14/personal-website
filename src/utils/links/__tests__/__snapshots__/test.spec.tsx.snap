// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`links headerLinks should match snapshot 1`] = `
[
  {
    "href": "/skills",
    "label": "Skills",
  },
  {
    "href": "/experiences",
    "label": "Experiences",
  },
  {
    "href": "/projects",
    "label": "Projects",
  },
]
`;

exports[`links socialMedias should match snapshot 1`] = `
[
  {
    "icon": <Linkedin />,
    "link": "https://www.linkedin.com/in/gabrielrapuccigonzalez/",
    "name": "LinkedIn",
  },
  {
    "icon": <MarkGithub />,
    "link": "https://github.com/gabrielrg14",
    "name": "GitHub",
  },
]
`;
